python_modules/dagster/dagster/core/events/__init__.py

```diff
@@ -882,6 +922,46 @@ def pipeline_canceled(
             ),
         )
 
+    @staticmethod
+    def step_worker_starting(
+        step_context: IStepContext,
+        message: str,
+        metadata_entries: List[MetadataEntry],
+    ) -> "DagsterEvent":
+        return DagsterEvent.from_step(
+            DagsterEventType.STEP_WORKER_STARTING,
+            step_context,
+            message=message,
+            event_specific_data=EngineEventData(
+                metadata_entries=metadata_entries, marker_start="step_process_start"
+            ),
```

Comment:
one thing to double-check is that we're ok having this share EngineEventData - I think this would be the first case of two different event types sharing an event_specific_data class (but I would argue that we could benefit from a bit more consolidation in our event path in general...)